Userland/Libraries/LibTLS/TLSv12.cpp

```diff
@@ -209,6 +233,19 @@ bool Context::verify_chain() const
         return false;
     }
 
+    if (!host.is_empty()) {
+        auto first_certificate = local_chain->first();
+        auto subject_matches = certificate_subject_matches_host(first_certificate, host);
+        if (!subject_matches) {
+            dbgln("verify_chain: First certificate does not match the hostname");
+            return false;
+        }
+    } else {
+        // FIXME: The host is taken from m_context.extensions.SNI, when is this empty?
```

Comment:
when SNI is disabled. as the comment up there mentioned, you should just pick a leaf cert and verify it - if the server accepts the hello with no SNI, all its certificates will be for one domain only.

Reply:
(FIXME is fine if you don't want to implement it ofc)

Reply:
`TLSv12::connect` is always calling `tls_socket->set_sni(host)`, which will set `m_context.extensions.SNI` to the server we are connecting with. So I think it is always set to some value. This might not be how we want to have it in the long run, but for now I think this FIXME is okay.

Reply:
There _is_ a "disable sni" option :P
Anyway, FIXME is fine

Reply:
Even with SNI disabled `TLSv12::connect` will still set `m_context.extensions.SNI`, so this should always work.

Anyway, I will leave the FIXME for now, if it causes problems in the future, I will try to address them.